### PR TITLE
Fix: V0.77 RO-2934 - Update DictMethodEntityInstanceHelper to properly parse beginning and ending residue IDs for CATH, SCOP, SCOP2, and ECOD annotations

### PR DESCRIPTION
**Pull request by piehld**
Specifically, authSeqBeg and authSeqEnd are returned as integers but must be compared as strings in pAuthAsymD.

### CODE REVIEW

File: rcsb/utils/dictionary/DictionaryApiProviderWrapper.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def __init__(self, cachePath, useCache=True, cfgOb=None, configName=None, **kwar
         self.__configName = configName
         self.__contentInfoConfigName = "content_info_helper_configuration"
         self.__dictLocatorMap = self.__cfgOb.get("DICT_LOCATOR_CONFIG_MAP", sectionName=self.__contentInfoConfigName)
+        # self.__dictLocatorMap: dictionary of configured dictionary name mappings, e.g.:
+        #   ordereddict([('pdbx_comp_model_core', ['PDBX_DICT_LOCATOR', 'RCSB_COMP_MODEL_DICT_LOCATOR', 'MA_DICT_LOCATOR']),
+        #                ('pdbx', ['PDBX_DICT_LOCATOR', 'RCSB_DICT_LOCATOR']),
+        #                ('pdbx_core', ['PDBX_DICT_LOCATOR', 'RCSB_DICT_LOCATOR', 'VRPT_DICT_LOCATOR']), ...
```
Comment (josemduarte) [inline rcsb/utils/dictionary/DictionaryApiProviderWrapper.py:50]: Is commenting this out intended? Is it related to this fix?

**Review by josemduarte — CHANGES_REQUESTED** [review]
Looks good thanks! Just one doubt in my comments

File: rcsb/utils/dictionary/DictionaryApiProviderWrapper.py (hunk 1 above)
Comment (piehld) [reply rcsb/utils/dictionary/DictionaryApiProviderWrapper.py:50]: Yes that was intentional--I was just adding documentation to provide an example of what that object looked like and contained. Not related to any fix--purely documentation purposes.

Comment (josemduarte) [reply rcsb/utils/dictionary/DictionaryApiProviderWrapper.py:50]: Ok, thanks for explaining

**Review by josemduarte — APPROVED** [review]
LGTM

**Comment by piehld** [comment]
One comment I'll add before merging: For future reference/interest, we may end up finding that it would be better to just return the ranges as strings from [EcodClassificationProvider](https://github.com/rcsb/py-rcsb_utils_struct/blob/master/rcsb/utils/struct/EcodClassificationProvider.py) (as well as Scop2Provider, and possibly others), rather than integers. However, since there may be other processes that expect integers from those classes, I've decided not to do that for now to avoid possibly causing other issues downstream. But, that would likely be the more appropriate and clean solution if it can be done without conflict.

**Comment by josemduarte** [comment]
Not sure I fully understand, but let me explain the agreed conventions behind our internal representation of residue indices: we aim at using label_seq_ids (which are integers, strictly from 1 to n of the full deposited sequence) as the internal representation for databases and APIs. It is only at display time (e.g. Protein Feature View) when we may display both the label_seq_id and auth_seq_id (strings, some times with insertion codes).

**Comment by piehld** [comment]
OK thank you, that makes sense. In that case, the proper solution may be to adjust how the sequence IDs are represented in the author-to-sequence ID mapping dictionary, rather than the IDs as represented in the annotations. That would involve modifying the code in [DictMethodCommonUtil](https://github.com/rcsb/py-rcsb_utils_dictionary/blob/28aebcd9858e1f9fbd1dd26254f4350015440fc2/rcsb/utils/dictionary/DictMethodCommonUtils.py#L814), in which the `getPolymerIdMap` method is used to return a mapping dictionary to [DictMethodEntityInstanceHelper](https://github.com/rcsb/py-rcsb_utils_dictionary/blob/master/rcsb/utils/dictionary/DictMethodEntityInstanceHelper.py) that represents the keyed-IDs as strings instead of integers.

**Pull request merged**